tests/random_test.py

```diff
@@ -2221,7 +2213,7 @@ def test_tile(self):
     self.check_against_reference(key_func, arr_func, key)
 
   def test_concatenate(self):
-    key = random.PRNGKey(123)
+    key = random.key(123)
```

Comment:
So long as we want to support both uint32 arrays (via `PRNGKey`) and new-style key arrays (via `key`), should we be testing both? Could we parameterize the tests to that end?

Reply:
Yeah, that would probably be a good idea longer-term, but maybe out of scope for this PR?

Reply:
Does this change not currently reduce the test surface too, though? The line change here subs a `PRNGKey` call for a `key` call. The former allows for both old- and new-style key arrays to be tested based on the `enable_custom_prng` flag. And we test both settings of that flag across our various CI runs.

Reply:
My intent here was to reduce redundant test surface, while increasing relevant test surface. This class contains tests meant to test special behavior of key arrays. It includes several tests that are skipped if `JAX_ENABLE_CUSTOM_PRNG=False`, as well as a number of tests of functionality like `device_put`, `expand_dims`, `broadcast_to`, etc. which for normal arrays have plenty of coverage elsewhere. That's the redundant test coverage that is removed by this change.

What is not covered elsewhere is calling these functions on `PRNGKeyArray` objects, when `JAX_ENABLE_CUSTOM_PRNG=False`. That's the additional test coverage enabled by changing `PRNGKey` to `key` here, and removing the various `@skipIf` annotations below.

Does that make sense?

Reply:
In addition to this, I think you're right that we should introduce some sort of parameterization to our random API tests that exercise both new and old keys, but that seems orthogonal to the goal of this PR.

Reply:
Looking at it, we already have such a parameterization via test class inheritance; e.g. https://github.com/google/jax/blob/06f76bc6bc75e9b68af42310ad08dea75294b1ca/tests/random_test.py#L2021-L2023

One issue with it is that it is skipped when custom PRNG is not enabled globally. We should remove that, but it will take some effort because the tests throughout use the global flag as a proxy for whether the key is a `PRNGKeyArray`, which does not always hold.

Reply:
Yep that all makes sense now. Thanks for walking me through it.

I agree we can and should clean up how the tests are "parameterized" for either old- or new-style, and we can do that next up.